Data-Science/MLflow/bike-sharing-mlflow.ipynb

```diff
@@ -27,7 +27,65 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Set Expriment"
+    "## User Authentication"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "EZAF_ENV = \"hpe-staging-ezaf\"\n",
+    "token_url = f\"https://keycloak.{EZAF_ENV}.com/realms/UA/protocol/openid-connect/token\"\n",
+    "\n",
+    "data = {\n",
+    "    \"username\" : input(\"Enter username.\"),\n",
+    "    \"password\" : input(\"Enter password.\"),\n",
+    "    \"grant_type\" : \"password\",\n",
+    "    \"client_id\" : \"ua-grant\",\n",
+    "}\n",
+    "\n",
+    "token_responce = requests.post(token_url, data=data, allow_redirects=True, verify=False)\n",
+    "\n",
+    "token = token_responce.json()[\"access_token\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import mlflow\n",
+    "import os\n",
+    "def set_exp(exp_name):\n",
+    "    \"\"\"\n",
+    "    This function is used to register an experiment in mlflow tracking service.\n",
+    "    The name of the experiment provided by the user will be set in mlflow.\n",
+    "    \"\"\"\n",
+    "    try:\n",
+    "        usr = \"hpedemo-user01\"\n",
+    "        os.environ['MLFLOW_TRACKING_INSECURE_TLS'] = 'true'\n",
+    "        os.environ[\"AWS_ACCESS_KEY_ID\"] = \"s3\"\n",
+    "        os.environ[\"AWS_SECRET_ACCESS_KEY\"] = token\n",
+    "        os.environ[\"MLFLOW_S3_ENDPOINT_URL\"] = 'http://local-s3-service.ezdata-system.svc.cluster.local:30000'\n",
+    "        os.environ['MLFLOW_S3_IGNORE_TLS'] = f\"true\"\n",
+    "        os.environ['MLFLOW_TRACKING_TOKEN'] = token\n",
```

Comment:
These all env variable should be set in jupyterhub notebook startscript, please check with Vova how we already did in previous release.

CC - @prakash

Reply:
Agreed, much of this is not needed and resolved in startscript. Syncing with Vova about where MLFLOW_TRACKING_TOKEN is set in the notebook, so that the secret access key can be set at the same time. I've pushed a commit that removes the auth block again

Reply:
yeah, we need to simplify these examples a bit. 
BTW - we put the token into AWS ACCESS KEY and we set SECRET KEY to string like "s3". The example I shared had these changes.

Reply:
Agreed - was referencing solution for auth in old version of the tutorial. 

Better solution is setting the values when the rest of the mlflow environment is configured https://github.com/mapr/private-kubeflow/blob/ezaf-fy23-q3/components/example-notebook-servers/jupyter/0_load_magics.py#L58C1-L58C1 .

Reply:
We have to set in jupyterhub notebook side, because other apps also can use these variable as env.

Reply:
@ask664 Correct. I feel the load magics function I've linked that should be the appropriate place, as it is already setting mlflow environment variables, including one that uses the JWT token.